### PR TITLE
fwupd: remove unused dependencies

### PR DESCRIPTION
**Pull request by r-vdp**
## Description of changes

Follow up to  #260048.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/firmware/fwupd/default.nix [hunk 1]
```diff
@@ -194,16 +192,13 @@ stdenv.mkDerivation (finalAttrs: {
...
     tpm2-tss
-    efivar
     fwupd-efi
     protobufc
     modemmanager
     libmbim
     libcbor
     libqmi
     xz # for liblzma
-  ] ++ lib.optionals haveDell [
```
Comment (AndersonTorres) [inline pkgs/os-specific/linux/firmware/fwupd/default.nix:205]: is haveDell used elsewhere

Comment (R-VdP) [reply pkgs/os-specific/linux/firmware/fwupd/default.nix:205]: Yeah it is
https://github.com/NixOS/nixpkgs/blob/cfed17c5397808b9f77a7377d8af24c9a332c908/pkgs/os-specific/linux/firmware/fwupd/default.nix#L224-L225

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/os-specific/linux/firmware/fwupd/default.nix (hunk 1 above)
Comment (amaxine) [reply pkgs/os-specific/linux/firmware/fwupd/default.nix:205]: Is dell still the right name to gate it under? I'm not sure of the original reasoning but a casual search points to synaptics hardware also being present in other manufacturers' hardware, but maybe the support for firmware updates is limited to Dell here.

**Comment by Nyabinary** [comment]
Why hasn't this been merged yet? cc @AndersonTorres

**Comment by AndersonTorres** [comment]
Merge conflicts.
Solve them before asking for merge.

**Comment by R-VdP** [comment]
This is my PR. I lost track of it honestly.

Part of it was merged recently with the latest fwupd version update.

I can rebase this later to fix the conflict, but I'm traveling so not sure when I'll have an opportunity to open my laptop.

**Pull request closed without merging**